docroot/modules/humsci/hs_events/modules/hs_events_importer/src/Form/EventsImporterForm.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Drupal\hs_events_importer\Form;
+
+use Drupal\Component\Utility\UrlHelper;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Class EventsImporterForm.
+ */
+class EventsImporterForm extends ConfigFormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'events_importer_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return ['hs_events_importer.settings'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $form = parent::buildForm($form, $form_state);
+    $urls = $this->config('hs_events_importer.settings')->get('urls');
+    $form['urls'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Urls'),
+      '#description' => $this->t('Leave empty to disable importer'),
+      '#default_value' => implode("\n", $urls),
+    ];
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    parent::validateForm($form, $form_state);
+
+    $urls = array_filter(explode("\n", str_replace("\r", '', $form_state->getValue('urls'))));
```

Comment:
```suggestion
    $urls = array_filter(explode(PHP_EOL, str_replace(PHP_EOL, '', $form_state->getValue('urls'))));
```

PHP_EOL should be used for cross-platform compatibility.

Reply:
```suggestion
    $urls = array_filter(explode(PHP_EOL, $form_state->getValue('urls')));
```

Reply:
PHP_EOL is only the `\n` i found that while testing that there was also a `\r` at the end of lines: 
`someurl\r\n
anotherurl`

Reply:
I replaced the \n, but i think we still need the \r replace.

Reply:
TIL: https://stackoverflow.com/questions/3997336/explode-php-string-by-new-line

Yeah, looks like the browser submission takes lead on the line endings.